sbin/prepareWorkspace.sh

```diff
@@ -113,6 +102,95 @@ checkoutAndCloneOpenJDKGitRepo() {
   cd "${BUILD_CONFIG[WORKSPACE_DIR]}"
 }
 
+# Checkout the required code to build from the given cached git repo
+# Set checkoutRc to result so we can retry 
+checkoutRequiredCodeToBuild() {
+  checkoutRc=1
+
+  cd "${BUILD_CONFIG[WORKSPACE_DIR]}/${BUILD_CONFIG[WORKING_DIR]}/${BUILD_CONFIG[OPENJDK_SOURCE_DIR]}"
+
+  echo "checkoutRequiredCodeToBuild:"
+  echo "  workspace = ${BUILD_CONFIG[WORKSPACE_DIR]}/${BUILD_CONFIG[WORKING_DIR]}/${BUILD_CONFIG[OPENJDK_SOURCE_DIR]}"
+  echo "  BUILD_VARIANT = ${BUILD_CONFIG[BUILD_VARIANT]}"
+  echo "  TAG = ${BUILD_CONFIG[TAG]}"
+  echo "  BRANCH = ${BUILD_CONFIG[BRANCH]}"
+
+  # Ensure commands don't abort shell
+  set +e
+  local rc=0
+
+  local tag="${BUILD_CONFIG[TAG]}"
+  if [ "${BUILD_CONFIG[BUILD_VARIANT]}" != "${BUILD_VARIANT_OPENJ9}" ]; then
+    git fetch --tags
+    rc=$?
+    if [ $rc -eq 0 ]; then
+      if git show-ref -q --verify "refs/tags/${BUILD_CONFIG[BRANCH]}"; then
+        echo "looks like the scm ref given is a valid tag, so treat it as a tag"
+        tag="${BUILD_CONFIG[BRANCH]}"
+        BUILD_CONFIG[TAG]="${tag}"
+      fi
+    else
+      echo "Failed cmd: git fetch --tags"
+    fi
+  fi
+
+  if [ $rc -eq 0 ]; then
+    if [ "${tag}" ]; then
+      echo "Checking out tag ${tag}"
+      git fetch origin "refs/tags/${tag}:refs/tags/${tag}"
+      rc=$?
+      if [ $rc -eq 0 ]; then
+        git checkout "${tag}"
```

Comment:
I'm torn between the explicit nature of pulling out the rc on a separate line each time (easy to follow) and its verboseness.  Worth combining the two in a single line?

Reply:
I will change to this:
```
git fetch origin "refs/tags/${tag}:refs/tags/${tag}" || rc=$?
```
If git fails, then rc gets set to failed $?